src/main/java/com/example/onlinebookstore/dto/user/LoginUserDto.java

```diff
@@ -1,12 +1,12 @@
 package com.example.onlinebookstore.dto.user;
 
-import jakarta.validation.constraints.NotNull;
 import lombok.Data;
+import lombok.NonNull;
 
 @Data
 public class LoginUserDto {
-    @NotNull
+    @NonNull
```

Comment:
According to documentation, lombok's annotation doesn't look like what you require.
https://projectlombok.org/features/NonNull
It's more looks like a kind of fail-fast approach with NPE when object has null value when it shouldn't.
Please use jararta one for DTO validations